### ISSUE TITLE
[Improvement] Consider publishing the Helm chart to a chart repository

### ISSUE THREAD

**Issue by chgl** [apache/incubator-kyuubi#6057]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, deploying via the Helm chart requires downloading the repository and installing from the local directory. Having the chart released to a repository would simplify installation significantly and make it easier to keep the chart up-to-date as a user. 

### How should we improve?

There is https://github.com/helm/chart-releaser which uses github pages and should be relatively straight-forward to implement.
As of recently, it's also possible to publish Helm charts as OCI artifacts https://helm.sh/docs/topics/registries/. This would allow hosting the chart on ghcr.io or Docker Hub (https://docs.docker.com/docker-hub/oci-artifacts/#push-a-helm-chart).

In my personal helm chart repo, I've been releasing charts using both methods for quite some time:  https://github.com/chgl/charts/blob/master/.github/workflows/release.yaml. I'd be happy to contribute a workflow if it makes sense. Things still to figure out:
1. How should the chart be versioned? - using the above mentioned methods, we could release a new version of the helm chart independent of the main kyuubi release. Everytime the chart is changed, the Chart.yaml version needs to be updated within the PR.
2. Should we add release notes for the chart itself? - The released charts can be synced to ArtifactHub which provides a nice UI for viewing releases and release notes from annotations: https://artifacthub.io/docs/topics/annotations/helm/

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

**Comment by sudohainguyen** [comment]
💯 agree and following

**Comment by yaooqinn** [comment]
cc @pan3793 @dnskr @zwangsheng

**Comment by pan3793** [comment]
I understand that we should do this, but there is a bit tricky issue blocking us to move forward.

TL;DR - how should the user configure Hadoop ecosystem configurations? Embedded large-XML content into YAML is not a good idea in my experience. You may find detail background from my previous comments, any suggestions are welcome.

- https://github.com/apache/kyuubi/pull/6046
- https://github.com/apache/kyuubi/pull/5633#discussion_r1383050820

**Comment by chgl** [comment]
Good points, I've left some comments! Publishing the chart in a repository doesn't prevent someone from still downloading and using the chart locally, though.

**Comment by dnskr** [comment]
> I understand that we should do this, but there is a bit tricky issue blocking us to move forward.
> 
> TL;DR - how should the user provide Hadoop ecosystem configurations? Embedding large-XML content into YAML is not a good idea in my experience. You can find detailed background from my previous comments, any suggestions are welcome.
> 
> * [[K8S][HELM] Add Hadoop configuration support #6046](https://github.com/apache/kyuubi/pull/6046)
> * [[K8S][HELM]Save ETCD certificate in a Kubernetes Secret #5633 (comment)](https://github.com/apache/kyuubi/pull/5633#discussion_r1383050820)

I'm working on `[Improvement] Define general Helm chart configuration approach` issue to initiate the discussion and address the problem. I'll create it in a 2-3 days.

**Comment by dnskr** [comment]
FYI @chgl
I created the issue https://github.com/apache/kyuubi/issues/6123.
